int32 data
string oddeven
